addons/account_edi_ubl_cii/models/account_move.py

```diff
@@ -48,6 +49,46 @@ def action_invoice_download_ubl(self):
             'target': 'download',
         }
 
+    def action_group_ungroup_lines_by_tax(self):
+        """
+        This action allows the user to reload an imported move, grouping or not lines by tax
+        """
+        self.ensure_one()
+        self._check_move_for_group_ungroup_lines_by_tax()
+
+        # Check if lines look like they're grouped
+        tax_ids = [str(line.tax_ids.ids) for line in self.invoice_line_ids if line.tax_ids]
+        lines_grouped = len(tax_ids) == len(set(tax_ids))
+
+        old_amount_untaxed, old_amount_total = self.amount_untaxed, self.amount_total
+
+        attachments = self.env['ir.attachment'].search([
+            ('res_model', '=', 'account.move'),
+            ('res_id', '=', self.id),
+        ], order='create_date')
+        if not attachments:
+            raise UserError(_("Cannot find the origin file, try by importing it again"))
+
+        files_data = self._to_files_data(attachments)
+        files_data.extend(self._unwrap_attachments(files_data))
+        file_data_groups = self._group_files_data_into_groups_of_mixed_types(files_data)
+        success = False
+
+        for file_data_group in file_data_groups:
+            self.invoice_line_ids = [Command.clear()]
+            self.with_context(ungroup_invoice_lines=lines_grouped, group_invoice_lines=not lines_grouped)._extend_with_attachments(file_data_group)
+            if self.amount_untaxed == old_amount_untaxed and self.amount_total == old_amount_total:
```

Comment:
Never ever compare floats directly like that. Use `currency.compare_amounts` instead